blockchain/validators/src/config.rs

```diff
@@ -0,0 +1,212 @@
+        ));
+    };
+
+    Ed25519Public::try_from(&bytes[..]).map_err(serde::de::Error::custom)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use tempfile::tempdir;
+
+    static INPUT_TOML: &str = r#"
+        [[node]]
+        message_signing_pub_key = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+        first_block_index = 0
+
+        [[node]]
+        message_signing_pub_key = "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+        first_block_index = 0
+        last_block_index = 10
+
+        [[node]]
+        message_signing_pub_key = "cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc"
+        first_block_index = 0
+        last_block_index = 10
+
+        [[node]]
+        message_signing_pub_key = "cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc"
+        first_block_index = 20
+    "#;
+
+    #[test]
+    fn config_from_path() {
+        let dir = tempdir().unwrap();
+        let toml_file = dir.path().join("testing.toml");
+        fs::write(&toml_file, INPUT_TOML).unwrap();
+
+        let cfg = KeyValidityConfig::load_from_path(&toml_file).unwrap();
+        assert_eq!(
+            cfg,
+            KeyValidityConfig {
+                node: vec![
+                    NodeKeyValidity::new(
+                        Ed25519Public::try_from(vec![0xAA_u8; 32]).unwrap(),
+                        0,
+                        None
+                    ),
+                    NodeKeyValidity::new(
+                        Ed25519Public::try_from(vec![0xBB_u8; 32]).unwrap(),
+                        0,
+                        10
+                    ),
+                    NodeKeyValidity::new(
+                        Ed25519Public::try_from(vec![0xCC_u8; 32]).unwrap(),
+                        0,
+                        10
+                    ),
+                    NodeKeyValidity::new(
+                        Ed25519Public::try_from(vec![0xCC_u8; 32]).unwrap(),
+                        20,
+                        None
+                    ),
+                ],
+            }
+        );
+    }
+
+    #[test]
+    fn unsupported_extension() {
+        let dir = tempdir().unwrap();
+        let poml_file = dir.path().join("testing.poml");
+        fs::write(&poml_file, INPUT_TOML).unwrap();
+
+        let result = KeyValidityConfig::load_from_path(&poml_file);
+        assert!(matches!(result, Err(Error::UnrecognizedExtension(_))));
+    }
+
+    #[test]
+    fn no_extension() {
+        let dir = tempdir().unwrap();
+        let file = dir.path().join("testing");
+        fs::write(&file, INPUT_TOML).unwrap();
+
+        let result = KeyValidityConfig::load_from_path(&file);
+        assert!(matches!(result, Err(Error::MissingExtension)));
+    }
+
+    #[test]
+    fn config_with_pem_path() {
+        let dir = tempdir().unwrap();
+        let config_path = dir.path().join("testing.toml");
+        let pem_path = dir.path().join("test.pem");
```

Comment:
Definitely not needed as part of this pr and maybe never needed.
Should this support files relative to the toml file?
right now this will most likely treat relative files as relative to the current working directory.

Reply:
Good idea! I'll make it relative to the config file.

Reply:
I couldn't find a not-too-hacky way to get the path from `load_from_path` into the custom deserialize method `deserialize_pem_or_hex`. I'm guessing operators will use absolute paths by default, and it seems OK to require absolute paths (or/and document that the paths are treated relative to CWD).

Reply:
Sounds good.

Again not saying to do this now, just more curious if you feel this falls into the "hacky" camp.  In the past in Python I've use a context manager to change the current working directory to the path of the config file.  Looks like there is https://docs.rs/chwd/0.2.0/chwd/struct.ChangeWorkingDirectory.html which does something similar.

Reply:
Interesting! I ended up using a different approach, where I defer parsing as `Ed25519Public` and track the dir passed to `load()`

Reply:
IDK I just talked with @jcape he mentioned there is some concerns about changing working directory in app.
it looks like it's not thread safe.

The delayed parsing is probably the safest, but definitely feels like more than needed at this time